homeassistant/components/modbus/climate.py

```diff
@@ -156,13 +161,24 @@ async def async_set_temperature(self, **kwargs):
         target_temperature = int(
             (kwargs.get(ATTR_TEMPERATURE) - self._offset) / self._scale
         )
```

Comment:
`target_temperature` is truncated to an `int` here. Should the `float(target_temperature)` on line 172 not use the untruncated value?

Reply:
Ups good catch corrected, a reason to add more tests :-) Thanks.